File: game-admin/src/main/java/com/dkm/config/TokenConfig.java
package com.dkm.config;


import org.jetbrains.annotations.NotNull;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.HandlerInterceptor;
import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry;

@Configuration
public class TokenConfig {
    protected void addInterceptors(@NotNull InterceptorRegistry registry) {
        registry.addInterceptor((HandlerInterceptor)this.tokenInterceptor()).addPathPatterns("/**");
    }

    public void addResourceHandlers(@NotNull ResourceHandlerRegistry registry) {
        registry.addResourceHandler("/statics/**").addResourceLocations("classpath:/statics/");
        registry.addResourceHandler("/swagger-ui.html").addResourceLocations("classpath:/META-INF/resources/");
        registry.addResourceHandler("/**").addResourceLocations("classpath:/static/");
        registry.addResourceHandler("/webjars/**").addResourceLocations("classpath:/META-INF/resources/webjars/");
    }

    @Bean
    @NotNull
    public AdminTokenInterceptor tokenInterceptor() {
        return new AdminTokenInterceptor();
    }

}

File: game-admin/src/main/java/com/dkm/config/TokenConfig.kt
package com.dkm.config

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.module.SimpleModule
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.http.converter.HttpMessageConverter
import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter
import org.springframework.web.servlet.config.annotation.InterceptorRegistry
import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurationSupport

@Configuration
class TokenConfig : WebMvcConfigurationSupport() {
    @get:Bean
    val tokenInterceptor: TokenInterceptor
        get() = TokenInterceptor()

    override fun addInterceptors(registry: InterceptorRegistry) {
        super.addInterceptors(registry)
        registry.addInterceptor(tokenInterceptor).addPathPatterns("/admin/config/**")
    }


}